docs/locks.rst

```diff
@@ -3,5 +3,73 @@
 
 .. versionadded:: 4.2
 
+Coordinate coroutines with synchronization primitives analogous to those the
+standard library provides to threads.
+
+*(Note that these primitives are not actually thread-safe and cannot be used in
+place of those from the standard library--they are meant to coordinate Tornado
+coroutines in a single-threaded app, not to protect shared objects in a
+multithreaded app.)*
+
+.. _the-wait-notify-pattern:
+
+The Wait / Notify Pattern
+=========================
+Tornado locks follow a "wait / notify pattern": one coroutine waits to be
```

Comment:
I'm not sure this is actually a helpful unifying principle. It makes sense for Condition and Event, but it doesn't really make sense for Lock or Semaphore, where acquire and release always occur in matched pairs. I'd probably just give each class its own examples instead of trying to extract common "patterns".